File: EntityFX.NetBenchamarks.Core/Generic/MathBase.cs
using System;
using System.Runtime.CompilerServices;

namespace EntityFX.NetBenchmark.Core.Generic
{
    public abstract class MathBase<TResult> : BenchmarkBase<TResult>
    {
        public MathBase()
        {
            Iterrations = 200000000;
            Ratio = 0.1;
        }

        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        protected static double DoMath(long i, double li)
        {
            double rev = 1.0 / (i + 1.0);
            return Math.Abs(i) * Math.Acos(rev) * Math.Asin(rev) * Math.Atan(rev) +
                Math.Floor(li) + Math.Exp(rev) * Math.Cos(i) * Math.Sin(i) * Math.PI + Math.Sqrt(i);
        }
    }
}

File: src/python/entityfx/math_base.py

from entityfx.benchmark_base import BenchmarkBase
from entityfx.writer import Writer
import math

class MathBase(BenchmarkBase):
    
    def __init__(self, writer: Writer, print_to_console : bool=True, is_enabled : bool=True) -> None:
        super().__init__(writer, print_to_console, is_enabled)
        self._iterrations = 200000000
        self.ratio = .5
    
    @staticmethod
    def _doMath(i : int, li : float) -> float:
        rev = 1.0 / (((i) + 1.0))
        return (math.fabs(i) * math.acos(rev) * math.asin(rev) * math.atan(rev) + math.floor(li) + math.exp(rev) * math.cos(i) * math.sin(i) * math.pi) + math.sqrt(i)